cpp/include/cudf/io/text/multistate.hpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace cudf {
+namespace io {
+namespace text {
+
+/**
+ * @brief represents a single (begin, end] pair of possible state transition history.
+ *
+ */
+struct multistate_segment {
+ public:
+  inline constexpr multistate_segment() : _data(0) {}
+  inline constexpr multistate_segment(uint8_t head, uint8_t tail)
```

Comment:
`inline constexpr` is redundant. Likewise, `inline` isn't needed on member functions.